cdf/cdf.js

```diff
@@ -114,7 +119,7 @@ class Cdf {
    * @return {number}
    */
   sampleSingle() {
-    const y = Math.random();
+    const y = random(min(this.ys), max(this.ys));
```

Comment:
this.ys should go from 0 to 1. This should probably be checked in a validation in the beginning.